tests/run/xml-interpolation/XmlQuote_1.scala

```diff
@@ -0,0 +1,52 @@
+import scala.quoted._
+import scala.tasty.Tasty
+
+case class Xml(parts: String, arg: Any)
+
+class XmlQuote(ctx: StringContext) {
+  rewrite def xml(args: => Any*): Xml = ~XmlQuote.impl('(ctx), '(args))
+}
+
+object XmlQuote {
+  implicit rewrite def XmlQuote(ctx: => StringContext): XmlQuote = new XmlQuote(ctx)
+
+  def impl(ctx: Expr[StringContext], args: Expr[Seq[Any]])(implicit tasty: Tasty): Expr[Xml] = {
+    import tasty._
+    import Term._
+
+    // for debugging purpose
+    def pp(tree: Tree): Unit = {
+      println(tasty.showSourceCode.showTree(tree))
+    }
+
+    def isStringConstant(tree: Term) = tree match {
+      case Literal(_) => true
+      case _ => false
+    }
+
+    // _root_.scala.StringContext.apply(("p0", "p1": scala.<repeated>[scala#Predef.String]))
+    val parts = ctx.toTasty match {
+      case Inlined(_, _,
+        Apply(
+          Select(Select(Select(Ident("_root_"), "scala", _), "StringContext", _), "apply", _),
+          List(Typed(Repeated(values), _)))) if values.forall(isStringConstant) =>
+        values.collect { case Literal(Constant.String(value)) => value }
+    case _ =>
+      ???
```

Comment:
This should only happen if you write explicitly something like 
```scala
val sc = new StringContext(...)
new XmlQuote(sc).xml(...)
```
you can emit an error in this case using `new scala.quoted.QuotedError(...)` which will be translated into a compilation error.

Reply:
Thanks. I think we are still missing the ability to report an error at a custom position

Reply:
Yes we are. Should not be hard to add.